tests/general/views/test_consolidated_profile_view.py

```diff
@@ -0,0 +1,566 @@
+        self.create_multiple_profile_categories(theme2)
+
+        response = self.get(
+            'themes-count', profile_id=self.profile.pk,
+            geography_code=self.geo1.code, extra={'format': 'json'}
+        )
+        self.assert_http_200_ok()
+
+        theme_data = response.data
+        assert len(theme_data) == 2
+        assert theme_data[0]["name"] == theme2.name
+        assert theme_data[1]["name"] == theme1.name
+
+    def test_subtheme_reoder(self):
+
+        theme1 = ThemeFactory(profile=self.profile, name="TH1", order=0)
+        theme2 = ThemeFactory(profile=self.profile, name="TH2", order=1)
+        pc1 = self.create_pc(theme1, order=1, label=F"pc_{theme1.name}")
+        pc2 = self.create_pc(theme1, order=0, label=F"pc_{theme1.name}")
+        pc3 = self.create_pc(theme2, order=1, label=F"pc_{theme2.name}")
+        pc4 = self.create_pc(theme2, order=0, label=F"pc_{theme2.name}")
+
+        response = self.get(
+            'themes-count', profile_id=self.profile.pk,
+            geography_code=self.geo1.code, extra={'format': 'json'}
+        )
+        self.assert_http_200_ok()
+
+        theme_data = response.data
+        assert theme_data[0]["subthemes"][0]['label'] == pc2.label
+        assert theme_data[0]["subthemes"][1]['label'] == pc1.label
+        assert theme_data[1]["subthemes"][0]['label'] == pc4.label
+        assert theme_data[1]["subthemes"][1]['label'] == pc3.label
+
+    def test_subtheme_reorder_after_new_data(self):
+        theme1 = ThemeFactory(profile=self.profile, name="TH1", order=0)
+        pc1 = self.create_pc(theme1, order=1, label=F"pc_{theme1.name}")
+        pc2 = self.create_pc(theme1, order=0, label=F"pc_{theme1.name}")
+
+        response = self.get(
+            'themes-count', profile_id=self.profile.pk,
+            geography_code=self.geo1.code, extra={'format': 'json'}
+        )
+        self.assert_http_200_ok()
+
+        theme_data = response.data
+        subthemes = theme_data[0]["subthemes"]
+        assert subthemes[0]['label'] == pc2.label
+        assert subthemes[1]['label'] == pc1.label
+
+        pc3 = self.create_pc(theme1, order=0, label=F"pc_{theme1.name}")
+        pc4 = self.create_pc(theme1, order=2, label=F"pc_{theme1.name}")
+        pc1.order = 1
+        pc1.save()
+        pc2.order = 3
+        pc2.save()
+
+        response = self.get(
+            'themes-count', profile_id=self.profile.pk,
+            geography_code=self.geo1.code, extra={'format': 'json'}
+        )
+        self.assert_http_200_ok()
+
+        theme_data = response.data
+        subthemes = theme_data[0]["subthemes"]
+        assert subthemes[0]['label'] == pc3.label
+        assert subthemes[1]['label'] == pc1.label
+        assert subthemes[2]['label'] == pc4.label
+        assert subthemes[3]['label'] == pc2.label
+
+
+class TestProfileHighlightData(ConsolidatedProfileViewBase):
+
+    def test_highlight_without_data(self):
```

Comment:
your name sounds like there's a highlight with no data

```suggestion
    def test_zero_highlights(self):
```

Reply:
Done